src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -158,9 +158,13 @@ package object csv {
       // does not have to happen in the inner loop.
       val schema = dataFrame.schema
       val formatForIdx = schema.fieldNames.map(fname => schema(fname).dataType match {
-        case TimestampType => (timestamp: Any) =>
-          dateFormatter.format(new Date(timestamp.asInstanceOf[Timestamp].getTime))
-        case DateType => (date: Any) => dateFormatter.format(date)
+        case TimestampType => (timestamp: Any) => {
+          if (timestamp == null) nullValue
```

Comment:
formatForIdx is returning a set of formatting functions for each column. Each of those functions takes as input the type of the data in the column and returns a string.  In the case of a timestamp column, a timestamp is the parameter and it returns a formatted datetime string.